src/test/java/com/karankumar/bookproject/ui/components/utils/ComponentUtilTest.java

```diff
@@ -0,0 +1,46 @@
+package com.karankumar.bookproject.ui.components.utils;
+
+import com.vaadin.flow.component.HasValue;
+import com.vaadin.flow.component.textfield.TextField;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ComponentUtilTest {
+
+    @Test
+    void shouldClearComponentFields() {
+        TextField bookTitle = new TextField("Book title");
+        HasValue[] components = {
+                bookTitle
+        };
+
+        ComponentUtil.clearComponentFields(components);
+
+        assertThat(bookTitle.getValue()).isEmpty();
+    }
+
+    @Test
+    void shouldNotClearComponentFieldsWhenComponentIsNull() {
+        TextField book = null;
+        HasValue[] components = {
+                book
+        };
+
+        ComponentUtil.clearComponentFields(components);
+
+        assertThat(book).isNull();
+    }
+
+    @Test
+    void shouldNotClearComponentFieldsWhenComponentIsEmpty() {
```

Comment:
Could you please change the test name to something like: `shouldNotThrowExceptionWhenClearingEmptyComponent`

Reply:
Done